src/DIRAC/Resources/IdProvider/DIRACCLIIdProvider.py

```diff
@@ -0,0 +1,20 @@
+""" IdProvider based on OAuth2 protocol
```

Comment:
What is the purpose of this IdProvider?

Reply:
it is used to create a session with the DIRAC Authorization Server from the CLI.

Reply:
IdPs are IMHO:
- EGI check-in
- IaM
- the DIRAC CS (??)
- [whatever gives you an Identity and that we want to support]

The DIRAC AS is instead for AuthZ (IdP is for AuthN?)

I might be missing something, but I'd like the flow to be clear.

Reply:
DIRAC AS acts as a proxy server for AuthN on IdP. That is, for a DIRAC client installation when the user does not have a certificate, the URL specified in the configuration to DIRAC AS and DIRAC AS for DIRAC Client acts as IdP, but eventually the user will be redirected to IAM / CheckIn.

Reply:
`DIRACCLIIdProvider` and `WebAppDIRACIdProvider` classes was removed. `OAuth2IdProvider` will be used instead for DIRAC clients initialisation.